src/web/docusaurus/docs/api-services/auth.md

```diff
@@ -1,3 +1,86 @@
+---
+sidebar_position: 5
+---
+
 # Authentication and Authorization
 
-Documentation in construction...
+## Summary
+
+SSO service is required for authentication. Telescope doesn't store user information itself, we just send it to Seneca's SSO. From that interaction, Telescope creates a token that will only work for a limited amount of time. The token is used in authenticating in various Telescope services, without a need to make requests to Seneca's servers every time. The same token system is also used throughout our Supabase back end.
+
+## Stages
+
+_This is what happens step by step in the SSO service:_
+
+- Request to log in is sent with the user entering their log in information into Telescope's front end.
+- Log in data is sent to Seneca's servers.
+- Microsoft SAML-2.
+
+  :::info
+
+  See more in [Not so simple SAML](https://blog.humphd.org/not-so-simple-saml/)
+
+  :::
+
+- Auth cookies are returned to SSO service.
+- SSO service creates a token. A token is basically JSON data.
+
+### Tokens
+
+_Token includes information such as:_
+
+1. Who issued it (Us - Telescope server)
+2. Who the audience is (For whom it is made and works for. So, Telescope servers)
+3. Who is the subject of the token (Some ID. In our case, a hashed user email)
+4. Claims. Data about the user.
+
+### Claims
+
+_Claims include following data:_
+
+1. Email
+2. First name
+3. Last name
+4. Display name (composite of first and last names)
+5. Roles
+
+### Roles
+
+_Roles describe permissions for certain actions_
+
+1. Seneca (Authenticated with Seneca SSO user)
+2. Telescope (User has a Telescope account)
+3. Admin (User is a Telescope admin)
+4. Service (User can use protected routes to communicate with another microservice, such as Satellite)
+
+   :::caution Important
+
+   Service role might be deprecated and no longer used. Satellite used to be a a separate repository, now that it isn't - the role might not be needed. It is unclear if it currently exists. Please create an issue to figure out this mystery. After you do - update this information.
+
+   :::
+
+## Proceeding stages
+
+- SSO service signs the token using a JWT secret, making it viable for a limited amount of time.
+
+:::info
+
+JWT is a string that has 3 parts. These parts are JavaScript objects made into a single line.
+Each part then base 64 encoded and is meant to be put in a URL. Parts are divided with `.` (dots)
```

Comment:
I don't quite understand this. The 3 parts are encoded with the base64 encoder?

Reply:
The information is encoded with base64. This is what was said in the [meeting](https://seneca.sharepoint.com/sites/Telescope/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FTelescope%2FShared%20Documents%2FGeneral%2FRecordings%2FJWT%20Secret%20in%20Telescope%2D20220406%5F135311%2DMeeting%20Recording%2Emp4&parent=%2Fsites%2FTelescope%2FShared%20Documents%2FGeneral%2FRecordings), specifically at 08:01.

Reply:
If you have a suggestion how to reword this better, make another comment please

Reply:
We can reword it to
` Each parts are encoded with the base64 encoder and then it is used in a URL`

Reply:
We can reword it to
` Each parts are encoded with the base64 encoder and then it is used in a URL`

Reply:
> We can reword it to ` Each parts are encoded with the base64 encoder and then it is used in a URL`

I'm not sure this wording is better, honestly